crates/matrix-sdk/src/encryption/recovery/mod.rs

```diff
@@ -527,6 +541,20 @@ impl Recovery {
         Ok(())
     }
 
+    async fn delete_all_known_secrets(&self) -> Result<()> {
+        for secret_name in Self::KNOWN_SECRETS {
+            let event_type = GlobalAccountDataEventType::from(secret_name.to_owned());
+            let content = SecretEventContent::new(Default::default());
+            let secret_content = Raw::from_json(
+                to_raw_value(&content)
+                    .expect("We should be able to serialize a raw empty secret event content"),
```

Comment:
Any chance we could use a plain `?` here, since this function returns a `Result` type anyways?

Reply:
As the recovery module has a specialized error type we can't just use `?` to convert the error. We either need to add a error variant for this or muck it somewhere into the `matrix_sdk::Error` type.

But I don't think it makes much sense, serializing the default event content should never fail.

Reply:
Makes sense to keep it like this, then :+1: